scene/gui/text_edit.cpp

```diff
@@ -1415,12 +1422,27 @@ void TextEdit::_notification(int p_what) {
 					ERR_CONTINUE(l < 0 || l >= completion_options.size());
 					Color text_color = cache.completion_font_color;
 					for (int j = 0; j < color_regions.size(); j++) {
-						if (completion_options[l].begins_with(color_regions[j].begin_key)) {
+						if (completion_options[l].insert_text.begins_with(color_regions[j].begin_key)) {
 							text_color = color_regions[j].color;
 						}
 					}
 					int yofs = (get_row_height() - cache.font->get_height()) / 2;
-					draw_string(cache.font, Point2(completion_rect.position.x, completion_rect.position.y + i * get_row_height() + cache.font->get_ascent() + yofs), completion_options[l], text_color, completion_rect.size.width);
+					Point2 title_pos(completion_rect.position.x, completion_rect.position.y + i * get_row_height() + cache.font->get_ascent() + yofs);
+#ifdef TOOLS_ENABLED
+					if (Engine::get_singleton()->is_editor_hint()) {
```

Comment:
Same as above here, no need to single this out to the editor / tools.

Reply:
If I understand correctly, you can simply remove the tools and editor hint checks and let `icon.is_valid()` be the determining factor.

Though I guess it's done this way so that you can allocate the icon width even if the icon is invalid, so that it's properly aligned?

Reply:
The editor checkings are removed now.

> Though I guess it's done this way so that you can allocate the icon width even if the icon is invalid, so that it's properly aligned?

The code completion feature can only be used in the editor for now (because related api are not exposed) so it should be fine.